src/storage-client/src/sink.rs

```diff
@@ -213,55 +169,46 @@ pub async fn ensure_kafka_topic(
         )
         .await
         .add_context("creating admin client failed")?;
-    // if either partition count or replication factor should be defaulted to the broker's config
-    // (signaled by a value of -1), explicitly poll the broker to discover the defaults.
+    // If either partition count or replication factor should be defaulted to the broker's config
+    // (signaled by a value of None), explicitly poll the broker to discover the defaults.
     // Newer versions of Kafka can instead send create topic requests with -1 and have this happen
     // behind the scenes, but this is unsupported and will result in errors on pre-2.4 Kafka.
-    if partition_count == -1 || replication_factor == -1 {
+    if partition_count.is_none() || replication_factor.is_none() {
```

Comment:
Commented on below, but the use of `Option<i32>` admits two ways to represent "broker default" (-1 and `None`), which means code might disagree on which to use.